todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/statistics/StatisticsFragment.java

```diff
@@ -59,24 +68,68 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onResume() {
         super.onResume();
-        mPresenter.subscribe();
+        bind();
     }
 
     @Override
     public void onPause() {
         super.onPause();
-        mPresenter.unsubscribe();
+        unbind();
     }
 
-    @Override
-    public void setProgressIndicator(boolean active) {
+    private void bind() {
+        mSubscription = new CompositeSubscription();
```

Comment:
You would do the `doOnSubscribe` and show the progress, in the fragment? if it's in the ViewModel, it can be unit tested. But, yes, trigger the progress in `doOnSubscribe` is a great idea, from both you and Tomek (above).